lib/jsonapi_rails/params_to_object.rb

```diff
@@ -134,12 +138,34 @@ def assign_attributes
     # that article to the *articles* relationship on the dynamically found
     # *person* object
     #
-    #
+    # @raise [UnknownRelationshipError] raised if the `resource` does not
+    #   define the given relationship
     def assign_relationships
       relationships.each do |rel|
         next if rel.blank?
+        json_api_relationship = resource.relationships.detect do |relationship|
+          relationship.name == rel[:name].to_s
+        end
+        unless json_api_relationship
+          message = "`#{resource.class}' does not have relationship " \
+                    "`#{rel[:name].to_s.gsub('=', '')}'"
+          raise UnknownRelationshipError.new(message)
+        end
+        relationship_value =
+          # JsonApi::Resources::ToManyRelationship implements #resource_objects
+          # JsonApi::Resources::ToOneRelationship implements #resource_object
+          # use these to identify what type of relationship we're working with
+          if json_api_relationship.respond_to?(:resource_objects)
```

Comment:
You should be able to call `relationship.cardinality` here. It'll return `:one` or `:many`